modules/avs/app/models/avs/v0/after_visit_summary.rb

```diff
@@ -40,6 +41,9 @@ def initialize(data)
         generated_date: data['generatedDate'],
         time_zone: data['data']['header']['timeZone']
       }
+      self.patient_info = {
+        smoking_status: data['data']['patientInfo']['smokingStatus']
```

Comment:
Chaining multiple bracket notations as in your code (data['data']['patientInfo']['smokingStatus']) could result in a NoMethodError if one of them is nil. I would recommend using dig here instead.

self.patient_info = {
  smoking_status: data.dig('data', 'patientInfo', 'smokingStatus')
}

Reply:
updated, thanks for the suggestion! 🙂